spec/models/exchange_rates/period_list_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'spec_helper'
+
+RSpec.describe ExchangeRates::PeriodList do
+  let(:year) { 2023 }
+  let(:api_host) { TradeTariffFrontend::ServiceChooser.api_host }
+  let(:response_headers) { { content_type: 'application/json; charset=utf-8' } }
+  let :response_data do
+    {
+      data: {
+        id: '2023-exchange_rate_period_list',
+        type: 'exchange_rate_period_list',
+        attributes: {
+          year:,
+        },
+        relationships: {
+          exchange_rate_periods: {
+            data: [
+              {
+                id: '2023-6-exchange_rate_period',
+                type: 'exchange_rate_period',
+                attributes: {
+                  month: 6,
+                  year:,
+                },
+              },
+              {
+                id: '2023-5-exchange_rate_period',
+                type: 'exchange_rate_period',
+                attributes: {
+                  month: 5,
+                  year:,
+                },
+              },
+            ],
+          },
+          exchange_rate_years: {
+            data: [
+              {
+                id: '2023-exchange_rate_year',
+                type: 'exchange_rate_year',
+                attributes: {
+                  year: 2023,
+                },
+              },
+              {
+                id: '2022-exchange_rate_year',
+                type: 'exchange_rate_year',
+                attributes: {
+                  year:,
+                },
+              },
+            ],
+          },
+        },
+      },
+      included: [
+        {
+          id: '2023-6-exchange_rate_period',
+          type: 'exchange_rate_period',
+          attributes: {
+            month: 6,
+            year:,
+          },
+        },
+        {
+          id: '2023-5-exchange_rate_period',
+          type: 'exchange_rate_period',
+          attributes: {
+            month: 5,
+            year:,
+          },
+        },
+        {
+          id: '2023-exchange_rate_year',
+          type: 'exchange_rate_year',
+          attributes: {
+            year:,
+          },
+        },
+        {
+          id: '2022-exchange_rate_year',
+          type: 'exchange_rate_year',
+          attributes: {
+            year: 2022,
+          },
+        },
+      ],
+    }
```

Comment:
this is quite a large data structure,
it make sense to me to put on a fixture `period_list_blablabla.json` and 
and load it using `file_fixture`:
```
def json_data(filename:)
  file_content = file_fixture("#{filename}.json").read
  JSON.parse(file_content, symbolize_names: true)
end
```

Reply:
I have limited the examples to have just 1 relationship to make this smaller. I have left it in the file as it makes it easier to read the test